Sources/SwiftInspectorVisitors/NestableTypeInfo.swift

```diff
@@ -22,26 +22,38 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-public struct ClassInfo: Codable, Equatable {
+protocol NestableTypeInfo {
```

Comment:
We create an internal protocol that all these three conform to in order to better enable working with all three in a compile-time safe manner.